views/components/config_card.erb

```diff
@@ -0,0 +1,95 @@
+<%# locals: (title:, config:, extra_class:, name:, errors:) %>
+
+<div class="bg-white rounded-lg shadow overflow-hidden mb-6 <%= extra_class %>">
+  <div class="px-6 py-4 border-b border-gray-200">
+    <h3 class="text-lg font-medium text-gray-900"><%= title %></h3>
+  </div>
+
+  <div class="divide-y divide-gray-200 w-full max-h-64 overflow-y-auto">
+    <% config.each_with_index do |(key, value), index| %>
+      <div class="px-6 py-2 hover:bg-gray-50 config-group group font-mono" data-config-id="<%= index %>">
+        <div class="flex items-center justify-between">
+          <div class="flex flex-wrap items-center config-entry">
+            <input
+              type="text"
+              class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500 mr-2 <%= (errors[key] && !errors[key].is_a?(Array)) ? "border-red-500" : "" %>"
+              value="<%= key %>"
+              name="<%= name %>_keys[]"
+              required
+            />
+            <input
+              type="text"
+              value="<%= value %>"
+              class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500 <%= errors[key].is_a?(Array) ? "border-red-500" : "" %>"
+              name="<%= name %>_values[]"
+            />
+            <% if errors[key] %>
+              <span class="error inline-block h-8 w-54 py-2 px-2 max-w-54 text-xs text-red-600 bg-red-100 rounded" aria-live="polite">
+                <% if errors[key].is_a?(Array) %>
+                  <% errors[key].each do |error| %>
+                    <%= error %><br/>
+                  <% end %>
+                <% else %>
+                  <%= errors[key] %>
+                <% end %>
+              </span>
+            <% end %>
+          </div>
+          <div class="min-w-16 flex items-center ml-2">
+            <%== part("components/button", type: "danger", text: "", icon: "hero-trash", extra_class: "delete-config-btn") if @edit_perm %>
+          </div>
+        </div>
+      </div>
+    <% end %>
+    <!-- Hidden row for generating new entries -->
+    <div class="px-6 py-2 hover:bg-gray-50 group config-placeholder-group font-mono hidden" data-config-id="">
+      <div class="flex items-center justify-between">
+        <div class="flex flex-wrap items-center config-entry">
+          <input
+            type="text"
+            class="border rounded py-1 focus:ring-blue-500 focus:border-blue-500 mr-2"
+            value=""
+            name="<%= name %>_keys[]"
```

Comment:
For hidden row, I would remove the name from the inputs, and add it via javascript when adding it to the form. This way the hidden row inputs are not submitted with the form.